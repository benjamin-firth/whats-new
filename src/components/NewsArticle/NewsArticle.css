.articleBox {
  justify-content: space-between;
  background-color: #FFFFFF;
  border: 5px solid #D4E9D9;
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  margin: 20px;
  padding: 5px;
  width: 330px;
}

.articlePic {
  width: 100%;
  height: 50%;
}

.articleLink {
  align-items: center;
  background-color: #252D33;
  border-radius: 5px;
  display: flex;
  height: 30px;
  justify-content: center;
  width: 100%;
}

.articleLink:hover { 
  background-color: #47A499;
}

h4 {
  color: #D4E9D9;
}