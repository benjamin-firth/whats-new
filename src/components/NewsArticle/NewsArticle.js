import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({ img, headline, description, url }) => {
  return (
    <article className="articleBox">
      <img src={img} alt='Article header background' className="articlePic"/>
      <h3>{headline}</h3>
      <hr/>
      <p>{description}</p>
      <a href={url}>Link to Article</a>
    </article>
  )
}

export default NewsArticle;