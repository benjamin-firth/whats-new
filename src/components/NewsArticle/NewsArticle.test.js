import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<NewsArticle 
      img={'image'}
      headline={'headline'}
      description={'my mock description'}
      url={'www.google.com'}
    />);
    expect(wrapper).toMatchSnapshot();
  })
})