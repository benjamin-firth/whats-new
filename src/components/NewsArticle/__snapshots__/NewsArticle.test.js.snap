// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot 1`] = `
<article
  className="articleBox"
>
  <img
    alt="Article header background"
    className="articlePic"
    src="image"
  />
  <h3>
    headline
  </h3>
  <hr />
  <p>
    my mock description
  </p>
  <a
    href="www.google.com"
  >
    <div
      className="articleLink"
    >
      <h4>
        Link to Article
      </h4>
    </div>
  </a>
</article>
`;
